plugin/DataTable.py

```diff
@@ -14,23 +14,33 @@
 import random
 import string
 import tempfile
+import urllib
 import websockets
 
 # mol 2d image drawing options
 Draw.DrawingOptions.atomLabelFontSize = 40
 Draw.DrawingOptions.dotsPerAngstrom = 100
 Draw.DrawingOptions.bondLineWidth = 8
 
-IS_DOCKER = os.path.exists('/.dockerenv')
 
 class DataTable(nanome.AsyncPluginInstance):
+
     @async_callback
     async def start(self):
         self.temp_dir = tempfile.TemporaryDirectory()
         self.temp_sdf = tempfile.NamedTemporaryFile(delete=False, suffix='.sdf', dir=self.temp_dir.name)
 
         self.url, self.https = self.custom_data
-        self.server_url = 'data-table-server' if IS_DOCKER else self.url
+        
+        # If url not resolvable by container, set server_url to data-table-server container name
+        protocol = 'https' if self.https else 'http'
+        try:
+            urllib.request.urlopen(f'{protocol}://{self.url}')
+        except urllib.error.URLError:
+            self.server_url = 'data-table-server'
+        else:
+            self.server_url = self.url        
```

Comment:
This should test connection to `'data-table-server'` first, then test `self.url` to prioritize local connection

Reply:
I'll switch it because it avoids the except clause in our current deploys, but long term the --url flag should be used. Hardcoding data-table-server is a kinda hacky/tech debt.